### ISSUE TITLE
[Feature Request] Enable hierarchical table view

### ISSUE THREAD

**Issue by gelicia** [grafana/grafana#83415]
### Discussed in https://github.com/grafana/grafana/discussions/57845

<div type='discussions-op-text'>

<sup>Originally posted by **lucaswhornsby** October 28, 2022</sup>
Building off this post in Grafana Community (https://community.grafana.com/t/hierarchy-in-grafana-table/53336), I would like to request the option to configure tables with a hierarchy such that individual rows can be comprised by sub-rows that can be expanded, and so forth. For an illustration, see below screenshot (from the original Grafana Community post). 

![image](https://user-images.githubusercontent.com/113460677/198706507-7ecb8252-d999-4a14-bfeb-5f446bd60134.png)

This feature would enable drill-down analysis that can bring significant value to multiple diverse Grafana use cases. 
</div>

**Comment by gato-mulatto** [comment]
A feature to have nested collapsible rows in the Grafana table visualization (as described above) would be hugely helpful. As requested by many in the Grafana discussions. It would be hugely helpful to myself as well.

**Comment by SqlBenjamin** [comment]
Agreed; this would be a valuable addition to tables. Please prioritize this one!

**Comment by leeoniya** [comment]
is the expectation here that the panel will have the full dataset returned by the datasource or are people expecting there to be some kind of additional querying when you click to expand (like a filesystem tree of unknown/huge size)?

**Comment by SqlBenjamin** [comment]
Great question. Personally, I like the idea of having the all the data to display in one query so the collapsing/expanding is fast without having to hit the backend as much. However, that means having a way to configure what is a parent and what's a child. Or perhaps a 'matrix' type of configuration - where you can create groups which would help define that relationship. 

Another reason I like this rather than a subquery (at least one that is defined/executed for each 'parent') is that you'd know when the data returns whether an item actually has children or not rather than having to click each item to know or having the query run for each row that is returned in order to return that info. For some data sources that may be okay, but for SQL that's not an efficient way to do it - avoid RBAR and do it in sets!

**Comment by mikhail-vl** [comment]
Hierarchical table views available in the community panel Business Table: https://grafana.com/grafana/plugins/volkovlabs-table-panel/?tab=overview. May help with discusses use cases.

![Image](https://github.com/user-attachments/assets/b0dfb5ac-5b34-46ae-82b4-a0fb9f575b21)

**Comment by SqlBenjamin** [comment]
I'm using Azure Managed Grafana and don't see that as an option there.

**Comment by gato-mulatto** [comment]
I long ago moved over to dash/plotly where this could easily be implemented. I do not believe it is available in Grafana. 
 
 
 
 
 
 
 
 
 SqlBenjamin  left a comment  (grafana/grafana#83415) 
 
 I'm using Azure Managed Grafana and don't see that as an option there.
